src/python/pants/core/goals/run_test.py

```diff
@@ -89,6 +91,10 @@ def single_target_run(
 ) -> Run:
     workspace = Workspace(rule_runner.scheduler, _enforce_effects=False)
 
+    def noop():
+        pass
+
+    monkeypatch.setattr("pants.engine.intrinsics.mark_nonrestartable", noop)
```

Comment:
This is the clunky part of this change, but not doing this monkeypatching will cause the test to fail with an obvious error, so at least it being missing is not subtle.

Note that the thing we're monkeypatching is the import of `mark_nonrestartable` by `intrinsics`.

Reply:
Is this potentially related to `SideEffecting._enforce_effects`?

https://github.com/pantsbuild/pants/blob/fa583d6b36df4c7bb1c560dc5429d1bfb474ba51/src/python/pants/engine/engine_aware.py#L102-L113

Reply:
Hah! I wasn't aware of `native_engine.task_side_effected()` but it is identical to my new `mark_nonrestartable()` and we don't need both, so will just use that.

Reply:
But that said, the monkeypatching is not due to this call, but the one in the new helper.